.github/workflows/build.yml

```diff
@@ -181,16 +181,25 @@ jobs:
           DEBUG: "pw:api"
           RELEASE: true # skip dev only test
 
+      - name: Save test artifacts - 1/2
+        if: ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
```

Comment:
No need to add this:

> failure
> Returns true when any previous step of a job fails. If you have a chain of dependent jobs, failure() returns true if any ancestor job fails.

Reply:
It's for early exit, otherwise the results from the second run is uploaded